### DISCUSSION TITLE
Attributes for blocks

### DISCUSSION THREAD

**Discussion by vsfeedback** [dotnet/csharplang#3290] [Language Ideas]
<p>I would love to be able to group C# members that use the same attributes, like:</p><pre>[SomeAttribute]
{
	public int Prop1 { get; set; }
	public string Prop2 { get; set; }
}
</pre><p>This would be equivalent of:</p><pre>[SomeAttribute]
public int Prop1 { get; set; }

[SomeAttribute]
public string Prop2 { get; set; }
</pre><p>Also, I think it might be useful for DependencyInjection scenarios, like:</p><pre>[DefaultConstructorInject]
{
	private readonly IService1 _service1 { get; private set; }
	private readonly IService2 _service2 { get; private set; }
}
</pre><p>The above would automatically generate below constructor (I hate writing it, as I feel like I'm repeating myself):</p><pre>public SomeClass(IService1 service1, IService2 service2)
{
	_service1 = service1;
	_service2 = service2;
}
</pre><p>Thanks!</p>

_This issue has been moved from https://developercommunity.visualstudio.com/content/idea/368725/attributes-for-blocks.html
VSTS ticketId: 718203_
_These are the original issue comments:_

Fiona Niu[MSFT] on 10/28/2018, 11:17 PM (4 days ago): <p>Thank you for taking the time to provide your suggestion. We will do some preliminary checks to make sure we can proceed further.&nbsp;You will hear from us in about a week on our next steps. </p>
Kendra Havens [MSFT] on 10/31/2018, 00:21 PM (45 hours ago): <p>Your suggestion has been queued up for prioritization.  Feature suggestions are prioritized based on the value to our broader developer community and the product roadmap. We may not be able to pursue this one immediately, but we will continue to monitor it up to 90 days for community input</p>

**Comment by Therzok** [comment]
Could be included in https://github.com/dotnet/roslyn/issues/6671 ?

**Comment by CyrusNajmabadi** [comment]
Not an IDE request, but a request for a new language feature.

**Comment by svick** [comment]
A very similar (and more general) proposal already exists at https://github.com/dotnet/csharplang/issues/818.
